packages/react-components/react-tree/src/components/TreeItem/useTreeItem.ts

```diff
@@ -77,20 +67,20 @@ export function useTreeItem_unstable<Value = string>(
         event,
         value,
         open: false,
-        type: treeDataTypes.arrowLeft,
+        type: treeDataTypes.ArrowLeft,
         target: event.currentTarget,
       });
     }
     if (!open && level > 1) {
-      return requestNavigation({ event, value, target: event.currentTarget, type: treeDataTypes.arrowLeft });
+      return requestNavigation({ event, value, target: event.currentTarget, type: treeDataTypes.ArrowLeft });
     }
   };
   const handleEnter = (event: React.KeyboardEvent<HTMLDivElement>) => {
     requestOpenChange({
       event,
       value,
-      open: isLeaf ? open : !open,
-      type: treeDataTypes.enter,
+      open: isBranch ? !open : open,
```

Comment:
as we discussed offline there are a lot of branching codepaths with the `isBranch` check which makes it hard to read and also a little dangerous maintenance wise.

You can do it in a follow up PR, but we should find some way to better organize the logic around branch/leaf tree items